interview-questions.md

```diff
@@ -4,49 +4,60 @@ Answer the following questions. First, without external resources. Challenge you
 
 
 
 2. What is a virtual DOM?
 
-  Your answer:
+  Your answer: I think that a virtual DOM is possibly a representation of the actual DOM. So instead of actually manipulating a DOM directly, we manipulate the virtual DOM
 
-  Researched answer:
+  Researched answer: A virtual DOM object is a representation of a DOM object, just like a "lightweight" copy. A virtual DOM has the same properties as a real DOM but it lacks the real DOM's ability to change whats on the screen. Manipulating the virtual DOM is far faster because nothing is drawn onscreen. So the example given by codeacademy is like editing the blueprints of a house rather than moving actual rooms in an actual house. So React compares when an update happens: it takes the updated virtual DOMs with a copy of the virtual DOM before the update happened, and it responds back by changing only those objects with a different virtual DOM, this process is called "diffing" React:
 
+1- The entire virtual DOM gets updated.
+2- The virtual DOM gets compared to what it looked like before you updated it. React figures out which objects have changed.
+3- The changed objects, and the changed objects only, get updated on the real DOM.
+4- Changes on the real DOM cause the screen to change.
 
 
 3. When creating a basic (stateless) class component in React, what are the necessary elements needed to display "Hello World" in the browser?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: I know that they cannot obviously contain state, and I think that a basic class component is called a functional component. So you need to just build a regular function for the (props) section?
 
+  Researched answer: So a basic class component do not manage their own state and dont have lifecycle methods (updating methods). We would need these necessary elements in a stateless component because they print out what is given to them via props, or the always render the same thing. I learned that stateless components dont even need props, but should have them to have child components ready to display.
+  a named function that returns as a Header, "Hello World"
+const Stateless = () => {
+  return(
+    <div>
+      <h1>Hello World</h1>
+    </div>
+    )
+}
 
 
 4. What is JSX?
 
-  Your answer:
+  Your answer: It is the language that React uses to communicate. It's like a hybrid ; part javascript, part HTML, part React's own logic.
 
-  Researched answer:
+  Researched answer: JSX is short for JavaScript XML. JSX is an expression which uses HTML statements within Javascript, so JSX allows us to write HTML in React. JSX converts HTML tags into react elements.
 
 
 
 5. What is state in React?
 
-  Your answer:
+  Your answer: State is a manipulated and updated variable used inside components in React for the use of making new data almost immediately present to alter how that specific component will change.
```

Comment:
Pretty much! It has to do with storing data/information to be use within the component or child components.